src/evaluator/evaluate.ts

```diff
@@ -156,10 +179,10 @@ export function evaluate({node, ...context}: EvaluateOptions): unknown {
       return node.attributes.reduce<Record<string, unknown>>((acc, attribute) => {
         switch (attribute.type) {
           case 'ObjectAttributeValue': {
-            const value = evaluate({...context, node: attribute.value})
-            if (value !== undefined) {
-              acc[attribute.name] = value
-            }
+            Object.defineProperty(acc, attribute.name, {
+              enumerable: true,
+              get: () => evaluate({...context, node: attribute.value}),
```

Comment:
lazy evaluation based on property access. what a great idea @judofyr

Reply:
Nice! I like this!